dataclients/kubernetes/kube.go

```diff
@@ -405,7 +422,7 @@ func (c *Client) ingressToRoutes(items []*ingressItem) []*eskip.Route {
 			host := []string{"^" + strings.Replace(rule.Host, ".", "[.]", -1) + "$"}
 
 			for _, prule := range rule.Http.Paths {
-				r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule)
+				r, servicesURLs, err = c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
```

Comment:
It is enough to pass the serviceURLs map it in the argument since map types are reference types. The additions inside the method will be carried over between calls.